spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/StreamParser.java

```diff
@@ -290,168 +286,54 @@ private SinkChannelNode eatSinkChannel() {
 		SinkChannelNode sinkChannelNode = null;
 		if (tokens.peek(TokenKind.GT)) {
 			Token gt = tokens.eat(TokenKind.GT);
-			ChannelNode channelNode = eatChannelReference(false);
+			ChannelNode channelNode = eatChannelReference();
+			if (channelNode == null) {
+				return null;
+			}
 			sinkChannelNode = new SinkChannelNode(channelNode, gt.startPos);
 		}
 		return sinkChannelNode;
 	}
 
 	/**
-	 * Enumeration of valid channel prefixes.
-	 */
-	enum ChannelPrefix {
-		queue, tap, topic;
-	}
-
-	/**
-	 * Return {@code true} if the provided string contains a valid
-	 * channel prefix.
-	 *
-	 * @param string string to check for a channel prefix
-	 * @return {@code true} if the provided string contains a channel prefix
-	 */
-	private boolean isLegalChannelPrefix(String string) {
-		return string.equals(ChannelPrefix.queue.toString()) ||
-				string.equals(ChannelPrefix.topic.toString()) ||
-				string.equals(ChannelPrefix.tap.toString());
-	}
-
-	/**
 	 * Return a {@link ChannelNode} for the token at the current position.
 	 * <p>
-	 * A channel reference is a colon separated list of identifiers that determine
-	 * the appropriate scope then a sequence of dot separated identifiers that
-	 * reference something within that scope.
-	 * <p>
-	 * Three types of top level prefix are supported for channels:
-	 * <ul>
-	 *     <li>queue</li>
-	 *     <li>topic</li>
-	 *     <li>tap</li>
-	 * </ul>
+	 * A channel reference is the label component when referencing a specific
+	 * module/label in a stream definition.
+	 *
 	 * Expected format:
-	 * {@code identifier [ ':' identifier ]* [ '.' identifier ]*}
+	 * {@code [ ':' identifier ]* [ '.' identifier ]*}
```

Comment:
this doesn't look valid now (shows 2 things with ZERO-or-more)... shouldn't it be something like below?:

```{@code ':' identifier ['.' identifier]*}```